code/modules/mod/mod_modules.dm

```diff
@@ -0,0 +1,616 @@
+	mod.slowdown -= slowdown_active
+	mod.wearer.update_equipment_speed_mods()
+	mod.wearer.update_gravity(mod.wearer.has_gravity())
+
+/obj/item/mod/module/holster
+	name = "MOD holster module"
+	desc = "A module that can instantly holster a gun inside the MOD."
+	module_type = MODULE_USABLE
+	complexity = 2
+	use_power_cost = 25
+	incompatible_modules = list(/obj/item/mod/module/holster)
+	cooldown_time = 0.5 SECONDS
+	var/obj/item/gun/holstered
+
+/obj/item/mod/module/holster/on_use()
+	. = ..()
+	if(!.)
+		return
+	if(!holstered)
+		var/obj/item/gun/holding = mod.wearer.get_active_held_item()
+		if(!holding)
+			to_chat(mod.wearer, "<span class='notice'>You aren't holding anything to holster!</span>")
+			return
+		if(!istype(holding) || holding.w_class > WEIGHT_CLASS_BULKY || holding.weapon_weight > WEAPON_MEDIUM)
+			to_chat(mod.wearer, "<span class='notice'>[holding] doesn't fit in the holster!</span>")
+			return
+		if(mod.wearer.transferItemToLoc(holding, src, FALSE, FALSE))
+			holstered = holding
+			to_chat(mod.wearer, "<span class='notice'>You holster [holstered].</span>")
+			playsound(src, 'sound/weapons/gun/revolver/empty.ogg', 100, TRUE)
+	else
+		if(mod.wearer.put_in_active_hand(holstered, FALSE, TRUE))
+			to_chat(mod.wearer, "<span class='notice'>You draw [holstered].</span>")
+			holstered = null
+			playsound(src, 'sound/weapons/gun/revolver/empty.ogg', 100, TRUE)
+
+/obj/item/mod/module/holster/on_uninstall()
+	if(holstered)
+		holstered.forceMove(drop_location())
+		holstered = null
+
+/obj/item/mod/module/tether
+	name = "MOD emergency tether module"
+	desc = "A module that can shoot an emergency tether to pull yourself towards an object in 0-G."
+	module_type = MODULE_ACTIVE
+	complexity = 3
+	use_power_cost = 50
+	incompatible_modules = list(/obj/item/mod/module/tether)
+	cooldown_time = 1.5 SECONDS
+
+/obj/item/mod/module/tether/on_use()
+	if(mod.wearer.has_gravity(get_turf(src)))
+		to_chat(mod.wearer, "<span class='warning'>You need to be in a 0-G environment to use [src]!</span>")
+		playsound(src, 'sound/weapons/gun/general/dry_fire.ogg', 25, TRUE)
+		return FALSE
+	return ..()
+
+/obj/item/mod/module/tether/on_select_use(mob/source, atom/target)
+	. = ..()
+	if(!.)
+		return
+	var/obj/projectile/tether = new /obj/projectile/tether
+	tether.preparePixelProjectile(target, mod.wearer)
+	tether.firer = mod.wearer
+	INVOKE_ASYNC(tether, /obj/projectile/proc/fire)
+
+/obj/projectile/tether
+	name = "tether"
+	icon_state = "tether"
+	icon = 'icons/obj/mod.dmi'
+	pass_flags = PASSTABLE
+	damage = 0
+	nodamage = TRUE
+	range = 10
+	hitsound = 'sound/weapons/batonextend.ogg'
+	hitsound_wall = 'sound/weapons/batonextend.ogg'
+	var/line
+
+/obj/projectile/tether/fire(setAngle)
+	if(firer)
+		line = firer.Beam(src, "line", 'icons/obj/mod.dmi')
+	..()
+
+/obj/projectile/tether/on_hit(atom/target)
+	. = ..()
+	if(firer)
+		firer.throw_at(target, 10, 1, firer, FALSE, FALSE, null, MOVE_FORCE_NORMAL, TRUE)
```

Comment:
what if the thing you hit is small, though? could you let us indiana jones whip an item closer to us and/or let us scorpion "GET OVER HERE!" people?

Reply:
this is a tether for emergencies, not a chain to kill people with